hbase-common/src/main/java/org/apache/hadoop/hbase/util/Bytes.java

```diff
@@ -1367,7 +1363,7 @@ int putShort(byte[] bytes, int offset, short val) {
       public UnsafeConverter() {}
 
       static {
-        if (UNSAFE_UNALIGNED) {
+        if (UnsafeAvailChecker.unaligned()) {
```

Comment:
My expectation here is by the time the code is JITted the JVM will be able to predict UnsafeAvailChecker.unaligned() is both monomorphic and returns the same value for every invocation. There will be a bit of extra code to branch (and de-opt) if the prediction fails, but I believe the branches will have the right prediction hints. The generated code will have the same performance characteristics as previous. If this is a concern, a jmh-based microbenchmark could confirm.